api/env.mli

```diff
@@ -36,85 +29,93 @@ val set_debug_mode : string -> unit
 
-  val define      : loc -> ident -> bool -> term -> term option -> unit
-  (** [define l id body ty] defined the symbol [id] of type [ty] to be an alias of [body]. *)
+val declare     : t -> loc -> ident -> Signature.staticity -> term -> unit
+(** [declare_constant env l id st ty] declares the symbol [id] of type [ty] and
+    staticity [st]. *)
 
-  val add_rules   : Rule.untyped_rule list -> (Subst.Subst.t * Rule.typed_rule) list
-  (** [add_rules rule_lst] adds a list of rule to a symbol. All rules must be on the
-      same symbol. *)
+val define      : t -> loc -> ident -> bool -> term -> term option -> unit
+(** [define env l id body ty] defined the symbol [id] of type [ty] to be an alias of [body]. *)
 
-  (** {2 Type checking/inference} *)
+val add_rules   : t -> Rule.untyped_rule list -> (Subst.Subst.t * Rule.typed_rule) list
+(** [add_rules env rule_lst] adds a list of rule to a symbol. All rules must be on the
+    same symbol. *)
 
-  val infer : ?ctx:typed_context -> term         -> term
-  (** [infer ctx term] infers the type of [term] given the typed context [ctx] *)
+(** {2 Type checking/inference} *)
 
-  val check : ?ctx:typed_context -> term -> term -> unit
-  (** [infer ctx te ty] checks that [te] is of type [ty] given the typed context [ctx] *)
+val infer : t -> ?ctx:typed_context -> term         -> term
+(** [infer env ctx term] infers the type of [term] given the typed context [ctx] *)
 
-  (** {2 Safe Reduction/Conversion} *)
-  (** terms are typechecked before the reduction/conversion *)
+val check : t -> ?ctx:typed_context -> term -> term -> unit
+(** [infer env ctx te ty] checks that [te] is of type [ty] given the typed context [ctx] *)
 
-  val reduction : ?ctx:typed_context -> ?red:(Reduction.red_cfg) -> term -> term
-  (** [reduction ctx red te] checks first that [te] is well-typed then reduces it
-      according to the reduction configuration [red] *)
+(** {2 Safe Reduction/Conversion} *)
+(** terms are typechecked before the reduction/conversion *)
 
-  val are_convertible : ?ctx:typed_context -> term -> term -> bool
-  (** [are_convertible ctx tl tr] checks first that [tl] [tr] have the same type,
-      and then that they are convertible *)
+val reduction : t -> ?ctx:typed_context -> ?red:(Reduction.red_cfg) -> term -> term
+(** [reduction env ctx red te] checks first that [te] is well-typed then reduces it
+    according to the reduction configuration [red] *)
 
-  val unsafe_reduction : ?red:(Reduction.red_cfg) -> term -> term
-  (** [unsafe_reduction red te] reduces [te] according to the reduction configuration [red] *)
+val are_convertible : t -> ?ctx:typed_context -> term -> term -> bool
+(** [are_convertible env ctx tl tr] checks first that [tl] [tr] have the same type,
+    and then that they are convertible *)
 
-end
+val unsafe_reduction : t -> ?red:(Reduction.red_cfg) -> term -> term
+(** [unsafe_reduction env red te] reduces [te] according to the reduction configuration [red] *)
```

Comment:
`unsafe` because it doesn't check whether the argument is well typed?  It would be sensible to say why it is unsafe in the documentation.

Reply:
Good point. I've added a comment to say that the term was not type checked.